teamcity: failed tests on master: acceptance/TestVersionUpgrade

The following tests appear to have failed:

[#634962](https://teamcity.cockroachdb.com/viewLog.html?buildId=634962):


```
--- FAIL: acceptance/TestVersionUpgrade (64.700s)
test_log_scope.go:81: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestVersionUpgrade276891811
test_log_scope.go:62: use -show-logs to present logs inline
--- FAIL: acceptance/TestVersionUpgrade: TestVersionUpgrade/runMode=local (64.700s)
version_upgrade_test.go:274: binary=v1.1.0
version_upgrade_test.go:274: cluster=1.0
version_upgrade_test.go:274: cluster=1.0-3
version_upgrade_test.go:274: binary=v1.1.1
version_upgrade_test.go:274: cluster=1.1
version_upgrade_test.go:274: binary=v2.0.0
version_upgrade_test.go:274: cluster=1.1-6
version_upgrade_test.go:274: cluster=2.0
version_upgrade_test.go:274: binary=source
version_upgrade_test.go:274: cluster=2.0-3
version_upgrade_test.go:191: 0: pq: value differs between gossip (2.0) and KV (2.0-3); try again later
------- Stdout: -------
CockroachDB node starting at 2018-05-01 13:01:48.322880817 +0000 UTC
build:      CCL v1.0.6 @ 2017/09/14 15:15:48 (go1.8.3)
admin:      http://127.0.0.1:38801
sql:        postgresql://root@127.0.0.1:44033?sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/1
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/1
status:     initialized new cluster
clusterID:  3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:     1
CockroachDB node starting at 2018-05-01 13:01:48.713359757 +0000 UTC
build:      CCL v1.0.6 @ 2017/09/14 15:15:48 (go1.8.3)
admin:      http://127.0.0.1:46805
sql:        postgresql://root@127.0.0.1:46829?sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/2
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/2
status:     initialized new node, joined pre-existing cluster
clusterID:  3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:     3
CockroachDB node starting at 2018-05-01 13:01:48.717297814 +0000 UTC
build:      CCL v1.0.6 @ 2017/09/14 15:15:48 (go1.8.3)
admin:      http://127.0.0.1:37723
sql:        postgresql://root@127.0.0.1:45035?sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/3
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/3
status:     initialized new node, joined pre-existing cluster
clusterID:  3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:     2
CockroachDB node starting at 2018-05-01 13:01:56.910840837 +0000 UTC (took 6.4s)
build:      CCL v1.1.0 @ 2017/10/12 14:50:18 (go1.8.3)
admin:      http://127.0.0.1:38709
sql:        postgresql://root@127.0.0.1:39709?application_name=cockroach&sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/1
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/1
status:     restarted pre-existing node
clusterID:  3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:     1
CockroachDB node starting at 2018-05-01 13:01:58.722333536 +0000 UTC (took 0.6s)
build:      CCL v1.1.0 @ 2017/10/12 14:50:18 (go1.8.3)
admin:      http://127.0.0.1:33821
sql:        postgresql://root@127.0.0.1:35075?application_name=cockroach&sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/2
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/2
status:     restarted pre-existing node
clusterID:  3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:     3
CockroachDB node starting at 2018-05-01 13:02:00.693083543 +0000 UTC (took 0.6s)
build:      CCL v1.1.0 @ 2017/10/12 14:50:18 (go1.8.3)
admin:      http://127.0.0.1:43435
sql:        postgresql://root@127.0.0.1:35103?application_name=cockroach&sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/3
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/3
status:     restarted pre-existing node
clusterID:  3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:     2
CockroachDB node starting at 2018-05-01 13:02:07.532585255 +0000 UTC (took 2.8s)
build:      CCL v1.1.1 @ 2017/10/19 15:31:46 (go1.8.3)
admin:      http://127.0.0.1:40671
sql:        postgresql://root@127.0.0.1:36667?application_name=cockroach&sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/1
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/1
status:     restarted pre-existing node
clusterID:  3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:     1
CockroachDB node starting at 2018-05-01 13:02:11.70323829 +0000 UTC (took 2.9s)
build:      CCL v1.1.1 @ 2017/10/19 15:31:46 (go1.8.3)
admin:      http://127.0.0.1:44245
sql:        postgresql://root@127.0.0.1:45619?application_name=cockroach&sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/2
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/2
status:     restarted pre-existing node
clusterID:  3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:     3
CockroachDB node starting at 2018-05-01 13:02:13.526167501 +0000 UTC (took 0.6s)
build:      CCL v1.1.1 @ 2017/10/19 15:31:46 (go1.8.3)
admin:      http://127.0.0.1:46621
sql:        postgresql://root@127.0.0.1:42055?application_name=cockroach&sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/3
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/3
status:     restarted pre-existing node
clusterID:  3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:     2
CockroachDB node starting at 2018-05-01 13:02:23.396032509 +0000 UTC (took 5.6s)
build:               CCL v2.0.0 @ 2018/04/03 20:56:09 (go1.10)
admin:               http://127.0.0.1:44991
sql:                 postgresql://root@127.0.0.1:43253?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/1
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/1/cockroach-temp516094315
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/1/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/1
status:              restarted pre-existing node
clusterID:           3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:              1
CockroachDB node starting at 2018-05-01 13:02:25.641356111 +0000 UTC (took 0.9s)
build:               CCL v2.0.0 @ 2018/04/03 20:56:09 (go1.10)
admin:               http://127.0.0.1:44751
sql:                 postgresql://root@127.0.0.1:44887?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/2
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/2/cockroach-temp396962543
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/2/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/2
status:              restarted pre-existing node
clusterID:           3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:              3
CockroachDB node starting at 2018-05-01 13:02:27.450073847 +0000 UTC (took 0.6s)
build:               CCL v2.0.0 @ 2018/04/03 20:56:09 (go1.10)
admin:               http://127.0.0.1:43457
sql:                 postgresql://root@127.0.0.1:34417?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/3
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/3/cockroach-temp074849815
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/3/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/3
status:              restarted pre-existing node
clusterID:           3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:              2
CockroachDB node starting at 2018-05-01 13:02:32.597499051 +0000 UTC (took 0.7s)
build:               CCL v2.1.0-alpha.20180416-493-g598a263 @ 2018/05/01 12:48:02 (go1.10)
admin:               http://127.0.0.1:36971
sql:                 postgresql://root@127.0.0.1:35455?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/1
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/1/cockroach-temp255035439
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/1/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/1
status:              restarted pre-existing node
clusterID:           3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:              1
CockroachDB node starting at 2018-05-01 13:02:38.516763575 +0000 UTC (took 4.7s)
build:               CCL v2.1.0-alpha.20180416-493-g598a263 @ 2018/05/01 12:48:02 (go1.10)
admin:               http://127.0.0.1:44943
sql:                 postgresql://root@127.0.0.1:44361?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/2
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/2/cockroach-temp790731473
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/2/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/2
status:              restarted pre-existing node
clusterID:           3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:              3
CockroachDB node starting at 2018-05-01 13:02:40.523028249 +0000 UTC (took 0.7s)
build:               CCL v2.1.0-alpha.20180416-493-g598a263 @ 2018/05/01 12:48:02 (go1.10)
admin:               http://127.0.0.1:44169
sql:                 postgresql://root@127.0.0.1:45365?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/3
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/3/cockroach-temp658941294
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/3/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/3
status:              restarted pre-existing node
clusterID:           3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:              2
test logs left over in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestVersionUpgrade276891811
--- FAIL: acceptance/TestVersionUpgrade (64.700s)
test_log_scope.go:81: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestVersionUpgrade276891811
test_log_scope.go:62: use -show-logs to present logs inline
--- FAIL: acceptance/TestVersionUpgrade: TestVersionUpgrade/runMode=local (64.700s)
version_upgrade_test.go:274: binary=v1.1.0
version_upgrade_test.go:274: cluster=1.0
version_upgrade_test.go:274: cluster=1.0-3
version_upgrade_test.go:274: binary=v1.1.1
version_upgrade_test.go:274: cluster=1.1
version_upgrade_test.go:274: binary=v2.0.0
version_upgrade_test.go:274: cluster=1.1-6
version_upgrade_test.go:274: cluster=2.0
version_upgrade_test.go:274: binary=source
version_upgrade_test.go:274: cluster=2.0-3
version_upgrade_test.go:191: 0: pq: value differs between gossip (2.0) and KV (2.0-3); try again later
------- Stdout: -------
CockroachDB node starting at 2018-05-01 13:01:48.322880817 +0000 UTC
build:      CCL v1.0.6 @ 2017/09/14 15:15:48 (go1.8.3)
admin:      http://127.0.0.1:38801
sql:        postgresql://root@127.0.0.1:44033?sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/1
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/1
status:     initialized new cluster
clusterID:  3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:     1
CockroachDB node starting at 2018-05-01 13:01:48.713359757 +0000 UTC
build:      CCL v1.0.6 @ 2017/09/14 15:15:48 (go1.8.3)
admin:      http://127.0.0.1:46805
sql:        postgresql://root@127.0.0.1:46829?sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/2
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/2
status:     initialized new node, joined pre-existing cluster
clusterID:  3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:     3
CockroachDB node starting at 2018-05-01 13:01:48.717297814 +0000 UTC
build:      CCL v1.0.6 @ 2017/09/14 15:15:48 (go1.8.3)
admin:      http://127.0.0.1:37723
sql:        postgresql://root@127.0.0.1:45035?sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/3
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/3
status:     initialized new node, joined pre-existing cluster
clusterID:  3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:     2
CockroachDB node starting at 2018-05-01 13:01:56.910840837 +0000 UTC (took 6.4s)
build:      CCL v1.1.0 @ 2017/10/12 14:50:18 (go1.8.3)
admin:      http://127.0.0.1:38709
sql:        postgresql://root@127.0.0.1:39709?application_name=cockroach&sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/1
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/1
status:     restarted pre-existing node
clusterID:  3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:     1
CockroachDB node starting at 2018-05-01 13:01:58.722333536 +0000 UTC (took 0.6s)
build:      CCL v1.1.0 @ 2017/10/12 14:50:18 (go1.8.3)
admin:      http://127.0.0.1:33821
sql:        postgresql://root@127.0.0.1:35075?application_name=cockroach&sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/2
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/2
status:     restarted pre-existing node
clusterID:  3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:     3
CockroachDB node starting at 2018-05-01 13:02:00.693083543 +0000 UTC (took 0.6s)
build:      CCL v1.1.0 @ 2017/10/12 14:50:18 (go1.8.3)
admin:      http://127.0.0.1:43435
sql:        postgresql://root@127.0.0.1:35103?application_name=cockroach&sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/3
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/3
status:     restarted pre-existing node
clusterID:  3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:     2
CockroachDB node starting at 2018-05-01 13:02:07.532585255 +0000 UTC (took 2.8s)
build:      CCL v1.1.1 @ 2017/10/19 15:31:46 (go1.8.3)
admin:      http://127.0.0.1:40671
sql:        postgresql://root@127.0.0.1:36667?application_name=cockroach&sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/1
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/1
status:     restarted pre-existing node
clusterID:  3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:     1
CockroachDB node starting at 2018-05-01 13:02:11.70323829 +0000 UTC (took 2.9s)
build:      CCL v1.1.1 @ 2017/10/19 15:31:46 (go1.8.3)
admin:      http://127.0.0.1:44245
sql:        postgresql://root@127.0.0.1:45619?application_name=cockroach&sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/2
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/2
status:     restarted pre-existing node
clusterID:  3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:     3
CockroachDB node starting at 2018-05-01 13:02:13.526167501 +0000 UTC (took 0.6s)
build:      CCL v1.1.1 @ 2017/10/19 15:31:46 (go1.8.3)
admin:      http://127.0.0.1:46621
sql:        postgresql://root@127.0.0.1:42055?application_name=cockroach&sslmode=disable
logs:       /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/3
store[0]:   path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/3
status:     restarted pre-existing node
clusterID:  3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:     2
CockroachDB node starting at 2018-05-01 13:02:23.396032509 +0000 UTC (took 5.6s)
build:               CCL v2.0.0 @ 2018/04/03 20:56:09 (go1.10)
admin:               http://127.0.0.1:44991
sql:                 postgresql://root@127.0.0.1:43253?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/1
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/1/cockroach-temp516094315
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/1/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/1
status:              restarted pre-existing node
clusterID:           3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:              1
CockroachDB node starting at 2018-05-01 13:02:25.641356111 +0000 UTC (took 0.9s)
build:               CCL v2.0.0 @ 2018/04/03 20:56:09 (go1.10)
admin:               http://127.0.0.1:44751
sql:                 postgresql://root@127.0.0.1:44887?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/2
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/2/cockroach-temp396962543
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/2/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/2
status:              restarted pre-existing node
clusterID:           3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:              3
CockroachDB node starting at 2018-05-01 13:02:27.450073847 +0000 UTC (took 0.6s)
build:               CCL v2.0.0 @ 2018/04/03 20:56:09 (go1.10)
admin:               http://127.0.0.1:43457
sql:                 postgresql://root@127.0.0.1:34417?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/3
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/3/cockroach-temp074849815
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/3/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/3
status:              restarted pre-existing node
clusterID:           3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:              2
CockroachDB node starting at 2018-05-01 13:02:32.597499051 +0000 UTC (took 0.7s)
build:               CCL v2.1.0-alpha.20180416-493-g598a263 @ 2018/05/01 12:48:02 (go1.10)
admin:               http://127.0.0.1:36971
sql:                 postgresql://root@127.0.0.1:35455?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/1
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/1/cockroach-temp255035439
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/1/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/1
status:              restarted pre-existing node
clusterID:           3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:              1
CockroachDB node starting at 2018-05-01 13:02:38.516763575 +0000 UTC (took 4.7s)
build:               CCL v2.1.0-alpha.20180416-493-g598a263 @ 2018/05/01 12:48:02 (go1.10)
admin:               http://127.0.0.1:44943
sql:                 postgresql://root@127.0.0.1:44361?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/2
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/2/cockroach-temp790731473
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/2/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/2
status:              restarted pre-existing node
clusterID:           3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:              3
CockroachDB node starting at 2018-05-01 13:02:40.523028249 +0000 UTC (took 0.7s)
build:               CCL v2.1.0-alpha.20180416-493-g598a263 @ 2018/05/01 12:48:02 (go1.10)
admin:               http://127.0.0.1:44169
sql:                 postgresql://root@127.0.0.1:45365?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestVersionUpgrade/runMode=local/3
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/3/cockroach-temp658941294
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/3/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster789109158/3
status:              restarted pre-existing node
clusterID:           3e97b3ea-b16d-4a24-be4f-5766bd494edc
nodeID:              2
test logs left over in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestVersionUpgrade276891811

```

Please assign, take a look and update the issue accordingly.

--- a/pkg/sql/show_cluster_setting.go
+++ b/pkg/sql/show_cluster_setting.go
@@ -18,10 +18,9 @@ import (
 	"bytes"
 	"context"
 	"fmt"
+	"math"
 	"strings"
-
-	"github.com/cockroachdb/cockroach/pkg/util/retry"
-	"github.com/pkg/errors"
+	"time"
 
 	"github.com/cockroachdb/cockroach/pkg/settings"
 	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
@@ -29,6 +28,8 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/types"
 	"github.com/cockroachdb/cockroach/pkg/sql/sqlbase"
 	"github.com/cockroachdb/cockroach/pkg/util/duration"
+	"github.com/cockroachdb/cockroach/pkg/util/retry"
+	"github.com/pkg/errors"
 )
 
 func (p *planner) showStateMachineSetting(
@@ -41,8 +42,10 @@ func (p *planner) showStateMachineSetting(
 	// immediately while at the same time guaranteeing that a node reporting a certain version has
 	// also processed the corresponding Gossip update (which is important as only then does the node
 	// update its persisted state; see #22796).
-	const maxAttempts = 10
-	if err := retry.WithMaxAttempts(ctx, retry.Options{}, maxAttempts, func() error {
+	retryCtx, cancel := context.WithTimeout(ctx, 20*time.Second)
+	defer cancel()
+	// The (slight ab)use of WithMaxAttempts achieves convenient context cancellation.
+	if err := retry.WithMaxAttempts(retryCtx, retry.Options{}, math.MaxInt32, func() error {
 		datums, err := p.ExtendedEvalContext().ExecCfg.InternalExecutor.QueryRow(
 			ctx, "read-setting",
 			p.txn,

